### PR TITLE
statx() syscall macro fix + reboot() handling improvements

### PR DESCRIPTION
**Pull request by poettering**
Primarily a fix for #8238. But then more stuff born out of it

### CODE REVIEW

File: src/core/emergency-action.c [hunk 1]
```diff
@@ -20,11 +20,12 @@
 ***/
 
 #include <sys/reboot.h>
-#include <linux/reboot.h>
 
 #include "bus-error.h"
 #include "bus-util.h"
 #include "emergency-action.h"
+#include "raw-reboot.h"
+#include "reboot-util.h"
```
Comment (yuwata) [inline src/core/emergency-action.c:28]: This is not necessary.

**Comment by poettering** [comment]
> Why don't you merge raw-reboot.h into reboot-util.h?

Well, they are semantically different things. raw-reboot.h follows raw-clone.h in its concepts: it includes a "raw" wrapper around a syscall, but its more than a pure wrapper, it actually will fill in redundant parameters for you, exactly like raw-clone.h, hence I figured it would make sense to also have a separate header for that, following the same logic and naming. This header doesn't do much, it just contains a static inline function definition, there's no raw-reboot.c file matching it.

reboot-util.[ch] otoh is different, it actually contains our code, adds concepts of our own, and adds real value, if you so will. As such it's more like the other xyz-util.[ch] files, hence I followed the scheme of it there.

File: src/core/emergency-action.c (hunk 1 above)
Comment (poettering) [reply src/core/emergency-action.c:28]: emergency-action.c calls update_reboot_parameter_and_warn() which is defined in reboot-util.h.

emergency-action.c also calls raw_reboot() which is defined in raw-reboot.h.

Hence it really needs both includes.

Comment (yuwata) [reply src/core/emergency-action.c:28]: Indeed. I thought that the functions provided by reboot-util.h are new. But this is not correct. Sorry for my misunderstanding.

**Comment by yuwata** [comment]
I see. Thank you for the explanation.

File: src/basic/missing_syscall.h [hunk 2]
```diff
@@ -415,9 +415,9 @@ static inline int missing_bpf(int cmd, union bpf_attr *attr, size_t size) {
 #if !HAVE_STATX
 #  ifndef __NR_statx
 #    if defined __i386__
-#      define __NR_bpf 383
+#      define __NR_statx 383
```
Comment (keszybz) [inline src/basic/missing_syscall.h:418]: Pffff.

File: src/basic/reboot-util.c [hunk 3]
```diff
@@ -0,0 +1,32 @@
+#include <errno.h>
```
Comment (keszybz) [inline src/basic/reboot-util.c:3]: This needs a SPDX line and possible a copyright line.

**Comment by poettering** [comment]
force pushed a new version with the spdx line added. 

I am convinced now that we shouldn't add new copyright lines/license blurbs to our files. I think git and SPDX are the better sources of information for this. Hence, let's make sure everything we add now comes with SPDX, and let's avoid all extra copyright/license blurbs (except maybe if we copy in somebody else's code).

Anyway, that's why I added the SPDX line now, but no copyright line.

**Pull request merged**